gh-123165: make `dis` functions render positions on demand (#123168)
python/cpython: Doc/library/dis.rst

      for jump targets and exception handlers. The ``-O`` command line
      option and the ``show_offsets`` argument were added.

   .. versionchanged:: 3.14

>       Added the :option:`-K <dis --show-positions>` command-line option

Reviewer: Why K? Could be P for positions or L for locations.

Author: I think I had "columns" in mind and then moved to 'K' instead of `C`. I'll use `P` for positions.

>    .. versionchanged:: 3.14
>       Added the :option:`-P <dis --show-positions>` command-line option
>       and the ``show_positions`` argument.

Reviewer: ```suggestion
   .. versionchanged:: 3.14
      The :option:`-P <dis --show-positions>` command-line option
      and the ``show_positions`` argument were added.
```

(to match the style of the previous entry).

   Show offsets of instructions.

.. cmdoption:: -P, --show-positions

>    Show positions of instructions.

Reviewer: ```suggestion
   Show positions of instructions in the source code.
```

   If *show_offsets* is ``True``, :meth:`.dis` will include instruction
   offsets in the output.

>    If *show_positions* is ``True``, :meth:`.dis` will include instruction
>    positions in the output.
> 

Reviewer: ```suggestion
   If *show_positions* is ``True``, :meth:`.dis` will include instruction
   source code positions in the output.

```

   .. versionchanged:: 3.13
      Added the *show_offsets* parameter.

   .. versionchanged:: 3.14
      Added the *show_positions* parameter.

.. function:: disassemble(code, lasti=-1, *, file=None, show_caches=False, adaptive=False)
              disco(code, lasti=-1, *, file=None, show_caches=False, adaptive=False,\
                    show_offsets=False, show_positions=False)

   Disassemble a code object, indicating the last instruction if *lasti* was
   provided.  The output is divided in the following columns:

>    #. the line number, for the first instruction of each line, or the
>       instruction positions if *show_positions* is true.

Reviewer: ```suggestion
   #. the source code location of the instruction. Complete location information is shown if
      *show_positions* is true. Otherwise (the default) only the line number is displayed.
```
